Submitted By: Joe Ciccone <jciccone@gmail.com>
Date: 16-09-2015
Initial Package Version: 2.6.18
Upstream Status: Unknown
Origin: Joe Ciccone
Description: This patch gives the ability to change the libdir.
             Updated for Iproute2-2.6.28 by Jim Gifford
             Updated for Iproute2-2.6.34 by Joe Ciccone
             Updated for Iproute2-2.6.37 by Joe Ciccone
             Renamed for Iproute2-2.6.38 by Jonathan Norman
             Rediffed for Iproute2-3.4.0 by William Harrington
             Rediffed for Iproute2-3.8.0 by William Harrington
             Renamed for Iproute2-3.12.0 by William Harrington
             Renamed for Iproute2-3.14.0 by William Harrington
             Rediffed for Iproute2-4.1.1 by William Harrington

diff -Naur iproute2-4.1.1.orig/Makefile iproute2-4.1.1/Makefile
--- iproute2-4.1.1.orig/Makefile	2015-07-06 21:57:34.000000000 +0000
+++ iproute2-4.1.1/Makefile	2015-09-16 23:57:03.117721908 +0000
@@ -13,7 +13,7 @@
 
 SHARED_LIBS = y
 
-DEFINES= -DRESOLVE_HOSTNAMES -DLIBDIR=\"$(LIBDIR)\"
+DEFINES= -DRESOLVE_HOSTNAMES -DLIBDIR=\"$(LIBDIR)\" -DARPDDIR=\"$(ARPDDIR)\"
 ifneq ($(SHARED_LIBS),y)
 DEFINES+= -DNO_SHARED_LIBS
 endif
diff -Naur iproute2-4.1.1.orig/include/iptables.h iproute2-4.1.1/include/iptables.h
--- iproute2-4.1.1.orig/include/iptables.h	2015-07-06 21:57:34.000000000 +0000
+++ iproute2-4.1.1/include/iptables.h	2015-09-16 23:57:03.118722035 +0000
@@ -4,8 +4,12 @@
 #include "iptables_common.h"
 #include "libiptc/libiptc.h"
 
+#ifndef LIBDIR
+#define LIBDIR "/usr/lib"
+#endif
+
 #ifndef IPT_LIB_DIR
-#define IPT_LIB_DIR "/usr/local/lib/iptables"
+#define IPT_LIB_DIR LIBDIR "/iptables"
 #endif
 
 #ifndef IPPROTO_SCTP
diff -Naur iproute2-4.1.1.orig/misc/arpd.c iproute2-4.1.1/misc/arpd.c
--- iproute2-4.1.1.orig/misc/arpd.c	2015-07-06 21:57:34.000000000 +0000
+++ iproute2-4.1.1/misc/arpd.c	2015-09-16 23:57:03.118722035 +0000
@@ -41,7 +41,7 @@
 int resolve_hosts;
 
 DB	*dbase;
-char	*dbname = "/var/lib/arpd/arpd.db";
+char   *dbname = ARPDDIR "/arpd.db";
 
 int	ifnum;
 int	*ifvec;
